WordPress/Classes/Services/HomepageSettingsService.swift

```diff
@@ -10,69 +10,81 @@ struct HomepageSettingsService {
 
     let blog: Blog
 
-    fileprivate let context: NSManagedObjectContext
+    fileprivate let coreDataStack: CoreDataStack
     fileprivate let remote: HomepageSettingsServiceRemote
     fileprivate let siteID: Int
 
-    init?(blog: Blog, context: NSManagedObjectContext) {
+    init?(blog: Blog, coreDataStack: CoreDataStack) {
         guard let api = blog.wordPressComRestApi(), let dotComID = blog.dotComID as? Int else {
             return nil
         }
 
         self.remote = HomepageSettingsServiceRemote(wordPressComRestApi: api)
         self.siteID = dotComID
         self.blog = blog
-        self.context = context
+        self.coreDataStack = coreDataStack
     }
 
     public func setHomepageType(_ type: HomepageType,
                                 withPostsPageID postsPageID: Int? = nil,
                                 homePageID: Int? = nil,
                                 success: @escaping () -> Void,
                                 failure: @escaping (Error) -> Void) {
+        var originalHomepageType: HomepageType?
+        var originalHomePageID: Int?
+        var originalPostsPageID: Int?
+        coreDataStack.performAndSave { context in
+            guard let blog = Blog.lookup(withObjectID: self.blog.objectID, in: context) else {
+                return
+            }
 
-        // Keep track of the original settings in case we need to revert
-        let originalHomepageType = blog.homepageType
-        let originalHomePageID = blog.homepagePageID
-        let originalPostsPageID = blog.homepagePostsPageID
+            // Keep track of the original settings in case we need to revert
+            originalHomepageType = blog.homepageType
+            originalHomePageID = blog.homepagePageID
+            originalPostsPageID = blog.homepagePostsPageID
 
-        switch type {
-        case .page:
-            blog.homepageType = .page
-            if let postsPageID = postsPageID {
-                blog.homepagePostsPageID = postsPageID
-                if postsPageID == originalHomePageID {
-                    // Don't allow the same page to be set for both values
-                    blog.homepagePageID = 0
+            switch type {
+            case .page:
+                blog.homepageType = .page
+                if let postsPageID = postsPageID {
+                    blog.homepagePostsPageID = postsPageID
+                    if postsPageID == originalHomePageID {
+                        // Don't allow the same page to be set for both values
+                        blog.homepagePageID = 0
+                    }
                 }
```

Comment:
Have you considered updating this to use `guard`? I find it clearer to read:

 ```suggestion
                guard let postsPageID = postsPageID else { return }
                
                blog.homepagePostsPageID = postsPageID
                
                guard let postsPageID == originalHomePageID else { return }
                
                // Don't allow the same page to be set for both values
                blog.homepagePageID = 0
```

Reply:
I don't think we should early return here, since the original code has two separate if statements.

Reply:
Ah, yes! I mustn't have noticed that in the diff.

Early return at `if let postsPageID` would result in the second check, `if let homePageID` not running 🤦‍♂️ 

Still, we _could_ use early return for the second, and last check.